addons/metadata.integral.artists/lib/scraper.py

```diff
@@ -0,0 +1,355 @@
+            result['biography'] = details['discogs']['biography']
+        discog = xbmcaddon.Addon().getSetting('discog')
+        if (discog in details) and ('albums' in details[discog]):
+            result['albums'] = details[discog]['albums']
+        genre = xbmcaddon.Addon().getSetting('genre')
+        if (genre in details) and ('genre' in details[genre]):
+            result['genre'] = details[genre]['genre']
+        style = xbmcaddon.Addon().getSetting('style')
+        if (style in details) and ('styles' in details[style]):
+            result['styles'] = details[style]['styles']
+        mood = xbmcaddon.Addon().getSetting('mood')
+        if (mood in details) and ('moods' in details[mood]):
+            result['moods'] = details[mood]['moods']
+        return result
+
+    def return_search(self, data):
+        for item in data:
+            listitem = xbmcgui.ListItem(item['artist'], thumbnailImage=item['thumb'], offscreen=True)
+            listitem.setProperty('artist.genre', item['genre'])
+            listitem.setProperty('artist.born', item['born'])
+            listitem.setProperty('relevance', item['relevance'])
+            url = {'artist':item['artist']}
+            if 'mbid' in item:
+                url['mbid'] = item['mbid']
+            if 'dcid' in item:
+                url['dcid'] = item['dcid']
+            xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=json.dumps(url), listitem=listitem, isFolder=True)
+
+    def return_nfourl(self, item):
+        url = {'artist':item['artist'], 'mbid':item['mbartistid']}
+        listitem = xbmcgui.ListItem(offscreen=True)
+        xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=json.dumps(url), listitem=listitem, isFolder=True)
+
+    def return_resolved(self, item):
+        url = {'artist':item['artist'], 'mbid':item['mbartistid']}
+        listitem = xbmcgui.ListItem(path=json.dumps(url), offscreen=True)
+        xbmcplugin.setResolvedUrl(handle=int(sys.argv[1]), succeeded=True, listitem=listitem)
+
+    def return_details(self, item):
+        listitem = xbmcgui.ListItem(item['artist'], offscreen=True)
+        if 'mbartistid' in item:
+            listitem.setProperty('artist.musicbrainzid', item['mbartistid'])
+        if 'genre' in item:
+            listitem.setProperty('artist.genre', item['genre'])
+        if 'biography' in item:
+            listitem.setProperty('artist.biography', item['biography'])
+        if 'styles' in item:
+            listitem.setProperty('artist.styles', item['styles'])
+        if 'moods' in item:
+            listitem.setProperty('artist.moods', item['moods'])
+        if 'instruments' in item:
+            listitem.setProperty('artist.instruments', item['instruments'])
+        if 'disambiguation' in item:
+            listitem.setProperty('artist.disambiguation', item['disambiguation'])
+        if 'type' in item:
+            listitem.setProperty('artist.type', item['type'])
+        if 'sortname' in item:
```

Comment:
Seems it allready has type and disambiguation properties, but needs "gender" too

Reply:
thx, added.